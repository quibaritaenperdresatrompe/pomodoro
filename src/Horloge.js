import { func, number, string } from "prop-types";
import { makeStyles } from "@material-ui/styles";
import React from "react";

import pad2Digits from "./pad2Digits";

const useStyles = makeStyles(() => ({
  time: {
    color: ({ color }) => color,
    cursor: "pointer",
    fontSize: () => {
      if (window.matchMedia("(min-width: 600px)").matches) {
        return 220;
      } else {
        return 100;
      }
    },
  },
  hours: {
    fontWeight: 700,
  },
}));

export default function Horloge({ color, onClick, time }) {
  const classes = useStyles({ color: !time ? "#d50c42" : color });
  return (
    <code className={classes.time} onClick={onClick}>
      <span className={classes.hours}>{pad2Digits(Math.floor(time / 60))}</span>
      :{pad2Digits(time % 60)}
    </code>
  );
}

Horloge.propTypes = {
  color: string,
  onClick: func,
  time: number,
};

Horloge.defaultProps = {
  color: "#000",
  onClick: Function.prototype,
  time: 0,
};
