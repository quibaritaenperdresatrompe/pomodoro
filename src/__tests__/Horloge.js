import { cleanup, render } from "@testing-library/react";
import React from "react";

import Horloge from "../Horloge";

describe(Horloge.name, () => {
  describe("on large screen, i.e. >= 600px", () => {
    beforeAll(() => {
      Object.defineProperty(window, "matchMedia", {
        writable: true,
        value: jest.fn(() => {
          return {
            matches: true,
            addEventListener: jest.fn(),
            removeEventListener: jest.fn(),
          };
        }),
      });
    });

    afterAll(cleanup);

    it("renders correctly", () => {
      const { asFragment } = render(<Horloge />);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe("on small screen, i.e. < 600px", () => {
    beforeAll(() => {
      Object.defineProperty(window, "matchMedia", {
        writable: true,
        value: jest.fn(() => {
          return {
            matches: false,
            addEventListener: jest.fn(),
            removeEventListener: jest.fn(),
          };
        }),
      });
    });

    afterAll(cleanup);

    it("renders correctly", () => {
      const { asFragment } = render(<Horloge />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
