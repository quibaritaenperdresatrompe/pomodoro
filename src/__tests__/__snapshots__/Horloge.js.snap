// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Horloge on large screen, i.e. >= 600px renders correctly 1`] = `
<DocumentFragment>
  <code
    class="makeStyles-time-1 makeStyles-time-3"
  >
    <span
      class="makeStyles-hours-2"
    >
      00
    </span>
    :00
  </code>
</DocumentFragment>
`;

exports[`Horloge on small screen, i.e. < 600px renders correctly 1`] = `
<DocumentFragment>
  <code
    class="makeStyles-time-4 makeStyles-time-6"
  >
    <span
      class="makeStyles-hours-5"
    >
      00
    </span>
    :00
  </code>
</DocumentFragment>
`;
